src/js/game/hud/parts/blueprint_placer.js

```diff
@@ -107,6 +107,11 @@ export class HUDBlueprintPlacer extends BaseHUDPart {
             const cost = blueprint.getCost();
             this.root.hubGoals.takeShapeByKey(blueprintShape, cost);
 
+            const leftover = this.root.hubGoals.getShapesStoredByKey(blueprintShape);
+            if (leftover < 50) {
+                this.root.hubGoals.putShapeByKey(blueprintShape, 50 - leftover);
+            }
+
```

Comment:
here it is

Reply:
So whenever you go below 50 shapes, you get it filled up again?

Reply:
yes